spec/controllers/institutions_controller_spec.rb

```diff
@@ -63,4 +108,32 @@
 
   end
 
+  describe "new_from_invite_data" do
+    it "with invite" do
+      invite = PendingInstitutionInvite.make(invited_user_email: user.email)
+      get :new_from_invite_data, {pending_institution_invite_id: invite.id}
+      expect(response).to be_success
+    end
+
+    it "without invite" do
+      expect do
+        get :new_from_invite_data
+      end.to raise_error(ActionView::MissingTemplate)
+    end
+
+    it "with accepted invite" do
+      invite = PendingInstitutionInvite.make(status: "accepted", invited_user_email: user.email)
+      expect do
+        get :new_from_invite_data, {pending_institution_invite_id: invite.id}
+      end.to raise_error(ActionView::MissingTemplate)
+    end
+
+    it "gets redirect from other page when pending invitation" do
+      dummy_institution = Institution.make # there must be some institution in the database for the redirect to trigger
```

Comment:
This is a bit uncool. Reason is here: https://github.com/instedd/cdx/blob/41c36c823840d4c126afa67f4d5bff3c81aec065/app/controllers/application_controller.rb#L63-L68
If there are no institutions, a different redirect rule catches first.

Reply:
I understand that. Still think it would be cleaner to act on a pending invitation first, if just to keep the test simple.
It's not practically relevant at all, so we can just leave it as is.